synapse/handlers/presence.py

```diff
@@ -1017,11 +1017,21 @@ def get_new_events(
             if from_key is not None:
                 from_key = int(from_key)
 
+            max_token = self.store.get_current_presence_token()
+            if from_key == max_token:
+                # This is necessary as due to the way stream ID generators work
+                # we may get updates that have a stream ID greater than the max
+                # token. This is usually fine, as it just means that we may send
+                # down some presence updates multiple times. However, we need to
+                # be careful that the sync stream actually does make some
```

Comment:
ok so I think this a bit confusing, since your fix is explicitly to *stop* the sync stream making progress.

perhaps you could spell out the situation as per https://github.com/matrix-org/synapse/pull/5507#issuecomment-505045974?